modules/repository-url/src/main/java/org/elasticsearch/repositories/url/URLRepository.java

```diff
@@ -82,21 +82,21 @@ public URLRepository(RepositoryMetaData metadata, Environment environment,
                          NamedXContentRegistry namedXContentRegistry) {
         super(metadata, environment.settings(), namedXContentRegistry);
 
-        if (URL_SETTING.exists(metadata.settings()) == false && REPOSITORIES_URL_SETTING.exists(settings) ==  false) {
+        if (URL_SETTING.exists(metadata.settings()) == false && REPOSITORIES_URL_SETTING.exists(environment.settings()) ==  false) {
```

Comment:
All of these repositories use `environment.settings()` instead of the "normal" settings object. This seems a bit fishy to be honest but it is what they have always done.

Reply:
Maybe! Or making it come back some other way. Or something. The trouble is that we enrich the settings that the rest of the application gets with other stuff so some stuff is just missing from `Environment.settings()` at it isn't obvious unless you've read a bunch of the startup code.